content/03_CreatingObservableSequences.md

```diff
@@ -0,0 +1,1274 @@
+
+Tasks come in two forms: `Task<T>`, which produces a result, and `Task`, which does not. But in Rx, there is only `IObservable<T>`—there isn't a no-result form. We've already seen this problem once before, when the `Observable.Start` method needed to be able to [adapt a delegate as an `IObservable<T>`](#from-delegates) even when the delegate was an `Action` that produced no result. The solution was to return an `IObservable<Unit>`, and that's also exactly what you get when you call `ToObservable` on a plain `Task`.
+
+The extension method is simple to use:
+
+```csharp
+Task<string> t = Task.Run(() =>
+{
+    Console.WriteLine("Task running...");
+    return "Test";
+});
+IObservable<string> source = t.ToObservable();
+source.Subscribe(
+    Console.WriteLine,
+    () => Console.WriteLine("completed"));
+source.Subscribe(
+    Console.WriteLine,
+    () => Console.WriteLine("completed"));
+```
+
+Here's the output.
+
+```
+Task running...
+Test
+completed
+Test
+completed
+```
+
+Notice that even with two subscribers, the task runs only once. That shouldn't be surprising since we only created a single task. If the task has not yet finished, then all subscribers will receive the result when it does. If the task has finished, the `IObservable<T>` effectively becomes a single-value cold observable.
+
+#### One Task per subscription
+
+There's a different way to get an `IObservable<T>` for a source. I can replace the first statement in the preceding example with this:
+
+```cs
+IObservable<string> source = Observable.FromAsync(() => Task.Run(() =>
+{
+    Console.WriteLine("Task running...");
+    return "Test";
+}));
+```
+
+Subscribing twice to this produces slightly different output:
+
+```
+Task running...
+Task running...
+Test
+Test
+completed
+completed
+```
+
+Notice that this executes the task twice, once for each call to `Subscribe`. `FromAsync` can do this because instead of passing a `Task<T>` we pass a callback that returns a `Task<T>`. It calls that when we call `Subscribe`, so each subscriber essentially gets their own task.
+
+If I want to use `async` and `await` to define my task, then I don't need to bother with the `Task.Run` because an `async` lambda creates a `Func<Task<T>>`, which is exactly the type `FromAsync` wants:
+
+```cs
+IObservable<string> source = Observable.FromAsync(async () =>
+{
+    Console.WriteLine("Task running...");
+    await Task.Delay(50);
+    return "Test";
+});
+```
+
+There is a subtle difference with this though. When I used `Task.Run` the lambda ran on a task pool thread from the start. But when I write it this way, the lambda will begin to run on whatever thread calls `Subscribe`. It's only when it hits the first `await` that it returns (and the call to `Subscribe` will then return), with the remainder of the method running on the thread pool.
```

Comment:
Since it produces output identical to the previous code I've just stated that this is the case, instead of actually showing the same output a 2nd time